lib/new.js

```diff
@@ -0,0 +1,687 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const puppeteer = require('puppeteer-core');const findChrome = require('./find_chrome');
+const {rpc} = require('../rpc');
+const debugApp = require('debug')('carlo:app');
+const debugServer = require('debug')('carlo:server');
+const {Color} = require('./color');
+const {HttpRequest} = require('./http_request');
+
+const fs = require('fs');
+const util = require('util');
+const {URL} = require('url');
```

Comment:
format: style mismatch:
```javascript
28|const util = require('util');
29|const {URL} = require('url');
30|const EventEmitter = require('events');
```
```suggestion
const {URL} = require('url');const EventEmitter = require('events');const fsReadFile = util.promisify(fs.readFile);let testMode = false;class App extends EventEmitter {/**
```

⏎ at column 30 should be removed.
Triggered rule # 279
```
  +1.roles not in {ARGUMENT, CALLEE}
	∧ +1.reserved not in {=, {}
	∧ +2.length ≥ 3
	∧ +2.reserved not in {(}
	∧ +2.roles not in {RIGHT}
	∧ +3.roles not in {EXPRESSION}
	∧ +5.internal_type not in {Identifier}
	∧ ^1.roles in {BLOCK} and not in {IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.roles not in {CALL}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -1.diff_offset ≤ 1
	∧ -2.diff_line = 0
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	∧ •••start_col ≥ 16
	⇒ y = ⏎
	Confidence: 0.782. Support: 62.
```
⏎ at column 40 should be removed.
Triggered rule # <N/A>
```
  +2.roles in {RIGHT}
	⇒ y = ␣
	Confidence: 0.997. Support: 195.
```
⏎⏎ at column 48 should be removed.
Triggered rule # 156
```
  +1.roles not in {ARGUMENT, MAP}
	∧ +1.reserved not in {]}
	∧ +2.roles not in {RIGHT}
	∧ ^1.roles not in {LITERAL, DECLARATION}
	∧ ^2.internal_type not in {ArrayExpression}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved = , and not in {:, .}
	∧ -2.diff_line = 0
	∧ -2.roles not in {EXPRESSION}
	∧ -3.length ≤ 1
	∧ -3.reserved not in {,}
	∧ -5.reserved not in {(}
	∧ -5.diff_col ≤ 27
	∧ •••start_col ≤ 66
	⇒ y = ⏎⏎
	Confidence: 0.500. Support: 3.
```
⏎⏎ at column 22 should be removed.
Triggered rule # 143
```
  +1.roles in {MAP} and not in {ARGUMENT}
	∧ +2.roles not in {RIGHT}
	∧ ^1.roles not in {DECLARATION}
	∧ ^2.roles not in {BINARY, MAP}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved = , and not in {:, .}
	∧ -2.diff_line = 0
	∧ -3.roles not in {VALUE}
	∧ -4.roles in {LITERAL}
	⇒ y = ⏎
	Confidence: 0.812. Support: 8.
```
⏎␣⁺␣⁺ at column 33 should be removed.
Triggered rule # 211
```
  +1.roles in {CALLEE} and not in {ARGUMENT}
	∧ +2.roles not in {RIGHT}
	∧ ^1.roles not in {OPERATOR, DECLARATION}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, .}
	∧ -2.diff_line = 0
	∧ -3.length ≥ 5
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	∧ -4.diff_col ≥ 8
	∧ •••start_line ≥ 38
	⇒ y = ⏎
	Confidence: 0.875. Support: 4.
```